### DISCUSSION TITLE
Mixing C# & F#. Source file | Project ; Any thoughts?

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/csharplang#6380] [Language Ideas]
We may use different tools for different languages, but they are going to work together.

We see different languages being mixed as a developer : Javascript + HTML | C# + Javascript + html (in Blazor) | C++ + Javascript (via emscripten) | English + Spanish in the south of the country.

For solving a same problem, different people with different backgrounds with different mindset may have different approaches. There is no need to create a seperate project with nearly identical names just because it's written in another language.

**Comment by Xyncgas** [comment]
Maybe it is better if I move this topic to the visual studio discussion

**Comment by HaloFour** [comment]
See: https://github.com/dotnet/roslyn/issues/21024

**Comment by huoyaoyuan** [comment]
I've examined this, but found it quite hard in practice. It becomes harder with source generators: every language needs some support of `partial`.

**Reply by En3Tho** [reply]
F# also has file order and it's not using Roslyn.

Also, I don't think .Net has something like "one compiler infrastructure to rule them all" with its own IR that can represent more operations than a single language can typically provide (for example, Rust like borrowing system). This way we could get same diagnostics and mix languages in same project easier.

Maybe it's impossible to build, I'm not sure. But sometimes I wish there were such a thing to make language creation for .Net easier.

**Reply by Xyncgas** [reply]
Well maybe that's what we get currently in 2020s, just gotta grow a couple years, mean while make some money go travel

**Reply by HaloFour** [reply]
Even if the compiler teams all wholeheartedly agreed and decided today to stop everything else to make this happen it wouldn't happen in the 2020s.  It involves starting over literally from scratch with every compiler for every language.

TL;DR, it's not going to happen.  The C# team considered it early in Roslyn's design and decided it wasn't worth the massive amount of effort that would need to go into it.

**Comment by saint4eva** [comment]
It is not useful and practical.

**Comment by Xyncgas** [comment]
would this make blazor in f# easier
